### PR TITLE
First PR into dev

### PR DESCRIPTION
**Pull request by jcwalker**
GPRegistryPolicyDsc - Initial PR into dev branch.
This resource module contains resources used to apply and manage local group policies by modifying the respective .pol file.

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/gpregistrypolicydsc/1)

### CODE REVIEW

File: Tests/Integration/MSFT_RegistryPolicyFile.config.ps1 [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            ValueData   = $node.FtValueData
+            ValueType   = $node.FtValueType
+        }
+
+        RefreshRegistryPolicy 'Integration_Test_RefreshAfter_ForcedTunneling'
+        {
+            Name = 'RefreshPolicyAfterForcedTunneling'
+        }
+    }
+}
```
Comment (bcwilhite) [inline Tests/Integration/MSFT_RegistryPolicyFile.config.ps1]: Needs new line...

**Comment by johlju** [comment]
I will continue with the review tomorrow . I should be able to get through all the files tomorrow. 🙂

**Comment by johlju** [comment]
@jcwalker I gone through all the files. Phew! 😄 Glad it was only two resource. 😆

**Comment by johlju** [comment]
Ignore the `# This does not work with semantic versions using dash, e.g. '1.0.0-alpha'.` in a review comment abaove. It was me not understanding how the prerelease tag is added to the module manifest.  
https://docs.microsoft.com/en-us/powershell/gallery/concepts/module-prerelease-support

```
(Test-ModuleManifest -Path .\*.psd1).PrivateData.PSData.Prerelease
```

Not sure it is needed at this time. My thought was we could publish a prerelease from the dev branch if one is present, and differ from one already deployed. A bit of logic needed for that. Maybe we can add that logic to the deploy step in the test framework instead (DscResource.Tests). Also @gaelcolas is working on a new build pipeline as well. So let us keep things simple to begin with. 🙂

**Comment by johlju** [comment]
@jcwalker Just two things left, after that I think it is good to go 🤔😄

**Comment by johlju** [comment]
Looks okay, but we seem to deploy a module with an automatic build number. Not sure where it comes from.
https://ci.appveyor.com/project/dscbot/gpregistrypolicydsc/builds/27494336#L1195

**Comment by johlju** [comment]
@jcwalker the deployed version number should be 1.0.0 as in the module manifest. Not sure why it used the AppVeyor build number `1.0.50.0`. I think we need to debug this a bit. I get on it a bit later today.

**Comment by jcwalker** [comment]
@johlju I believe it comes from the appveyor.yml on this [line](https://github.com/dsccommunity/GPRegistryPolicyDsc/blob/c8298e2db2327c746581c38a47656e2a91a29c3c/appveyor.yml#L5)

**Comment by johlju** [comment]
Yes agree, but don't understand how `Publish-Module` picks it up as the version and not the version inside the module manifest? 🤔 DO it get put into a environment variable that `Publish-Module` sees? 🤔

**Comment by jcwalker** [comment]
@johlju I thought Publish-Module takes the version in the manifest.

**Comment by johlju** [comment]
Yes it does, and it seems the module manifest changes when the CI runs. So it's okay. 
But do we already have code that changes the manifest? I did not know that. Where do we set this, is it in the test framework? 🤔 Looking further.

My debug run: 
https://ci.appveyor.com/project/johlju/gpregistrypolicydsc#L1196

RDP:ing into the build worker, running `Publish-Module` manually result in the same.

```powershell
PS > Publish-Module -Path . -NuGetApiKey any -WhatIf
What if: Performing the operation "Publish-Module" on target "Version '1.0.1.0' of module 'GPRegistryPolicyDsc'".
```
Looking at the manifest it actually do have that number!  😄 
```
@{
  # Version number of this module.
  moduleVersion = '1.0.1.0'

  ...
}
```

**Comment by johlju** [comment]
It seems it is set here in the `Invoke-AppveyorAfterTestTask` function in the test framework.

https://github.com/PowerShell/DscResource.Tests/blob/175d755b5720f61059171b991eb49aaeea795702/AppVeyor.psm1#L1072-L1078

But it does not use `Update-ModuleManifest`, so we need to have to look into that later too when building the deploy function.

**Comment by johlju** [comment]
Working on a workaround...

**Comment by johlju** [comment]
We need to get the correct version from the module manifest before the Invoke-AppVeyorAfterTestTask runs in the After Test step, then set back the correct version into the module manifest. If we do this in the deploy step we don’t know what the correct version was. 

We could of course get the version here, then set it back in the deploy step. Up to you :)

**Comment by jcwalker** [comment]
I'll use your suggestion.  I just wanted to make sure it was correct.

**Comment by johlju** [comment]
@jcwalker Do you make the PR from dev to master? Lets try to release this and see if everything works. 😄

**Pull request merged**